odxtools/compumethods/createanycompumethod.py

```diff
@@ -76,7 +76,7 @@ def _parse_compu_scale_to_linear_compu_method(
     )
     if internal_upper_limit is None:
         if not is_scale_linear:
-            internal_upper_limit = Limit(float("inf"), IntervalType.INFINITE)
+            internal_upper_limit = Limit(0, IntervalType.INFINITE)
```

Comment:
Sure about this?

Reply:
pretty much: it is impossible to convert an infinite `float` to `int` (try `int(float("inf"))`), and `Limit.complies_to_{upper,lower}()` ignore the value for `IntervalType.INFINITE` anyway...

Reply:
what about signed int, 0 is not the min value in that case

Reply:
the problem with this is that in python integer classes are arbitrary length, so there is no analogon to INT_MAX in C...

Reply:
The trick is knowing the type bit_length
```python
# bit length
bit_length = 4

def unsigned_to_signed(value):
    sign_mask = 1 << (bit_length - 1));
    if (sign_mask & raw) == 0:
        return value
    return (value ^ sign_mask) - sign_mask

max_unsigned = (1 << bit_length) - 1 # 15
min_unsigend = 0

min_signed = unsigned_to_signed(1 << (bit_length - 1)) # -8
max_signed =unsigned_to_signed((1 << (bit_length - 1)) - 1) # 7
```

Reply:
what would be the point? IMO, passing `None` would better, but I think that allowing `None` for the value would regress the normal cases (as I see it, infinite limits are a corner case)...